app/services/complaint_case_updater_service.rb

```diff
@@ -0,0 +1,37 @@
+class ComplaintCaseUpdaterService < CaseUpdaterService
+  attr_reader :result
+
+  INITAL_CASE_STATUS = 'to_be_assessed'
+
+  def call
+    reset_status_on_complaint_type_change
+    super
+  end
+
+  private
+
+  def reset_status_on_complaint_type_change
+    if current_case_type_doesnt_match_selected_case_type 
+      reset_case_status_to_default_state
+    end
+  end
+
+  def reset_case_status_to_default_state
+      @kase.current_state = INITAL_CASE_STATUS
+  end
+
+  def current_case_type_doesnt_match_selected_case_type
+    case_type = @kase.complaint_type
+    params_case_type = @params['complaint_type']
+
+    complaint_type_mapping(case_type) != params_case_type && params_case_type.present?
+  end
+
+  def complaint_type_mapping(case_complaint_type)
+    {
+      'Litigation': 'litigation_complaint',
+      'ICO': 'ico_complaint',
+      'Standard': 'standard_complaint'
+    }[case_complaint_type.to_sym]
+  end
```

Comment:
This mapping is not necessary and also a little risky as the wording may changes and it is maintained in en.yaml,  if the @kase is here is decorator class of OffenderComplaint model,  you can retrieve the value of complaint_type by using @kase.object.complaint_type.